phoebe/backend/backends.py

```diff
@@ -297,14 +301,16 @@ def _extract_from_bundle(b, compute, dataset=None, times=None,
                     info['mesh_kinds'] = [b.filter(dataset=ds, context='dataset', **_skip_filter_checks).kind for ds in info['mesh_datasets']]
 
                 if by_time:
-                    for time_ in this_times:
+                    for i, time_ in enumerate(this_times):
+                        info['original_index'] = i
+                        info_ = info.copy()
```

Comment:
this still concerns me a little - does this work well with plotting and animating or would we need to update that logic as well?  If we do decide to go in this direction, we probably should change the implementation for line profiles as well 🤔

Reply:
Yes and no -- the plotting works, but a plot 'vises' vs. 'u' (and line style) is not sorted according to 'u', so it ends up with criss-crossing lines. Do you know why, please? A gnuplot script this plots correctly...